GeometryDashAPI/Serialization/TypeDescriptor.cs

```diff
@@ -236,13 +234,8 @@ private static IEnumerable<MemberInfo> GetPropertiesAndFields(Type type)
         {
             foreach (var property in type.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic))
                 yield return property;
-            var current = type;
-            while (current != null && current != typeof(object))
-            {
-                foreach (var field in current.GetFields(BindingFlags.Instance | BindingFlags.NonPublic))
-                    yield return field;
-                current = current.BaseType;
-            }
+            foreach (var field in type.GetFields(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public))
+                yield return field;
```

Comment:
I've rollback this, because `GetFields` doesn't return private field from inherited classes  
Also added test for this `TypeDescriptorTests.PrivateFieldFromInheritedClass`

Reply:
Ah, i see, i thought private field form base class can be found using `GetFieds(NonPublic)`, my mistake.

Reply:
It’s ok, I thought too at first time xd